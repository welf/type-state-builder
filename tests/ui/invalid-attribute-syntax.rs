use type_state_builder::TypeStateBuilder;

#[derive(TypeStateBuilder)]
struct InvalidAttribute {
    // This should be an error: invalid attribute syntax
    #[builder(invalid_attribute)]
    name: String,

    email: String,
}

fn main() {}
