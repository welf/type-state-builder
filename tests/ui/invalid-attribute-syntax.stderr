error: Unknown builder attribute. Supported attributes: required, setter_name, setter_prefix, default, skip_setter, impl_into
 --> tests/ui/invalid-attribute-syntax.rs:6:15
  |
6 |     #[builder(invalid_attribute)]
  |               ^^^^^^^^^^^^^^^^^

error[E0599]: no function or associated item named `builder` found for struct `InvalidAttribute` in the current scope
  --> tests/ui/invalid-attribute-syntax.rs:13:39
   |
4  | struct InvalidAttribute {
   | ----------------------- function or associated item `builder` not found for this struct
...
13 |     let _instance = InvalidAttribute::builder()
   |                                       ^^^^^^^ function or associated item not found in `InvalidAttribute`
